pkg/config/install_requirements.go

```diff
@@ -425,24 +425,28 @@ func (c *RequirementsConfig) IsLazyCreateSecrets(flag string) (bool, error) {
 
 // addDefaults lets ensure any missing values have good defaults
 func (c *RequirementsConfig) addDefaults() {
-	if c.Webhook == WebhookTypeNone {
-		c.Webhook = WebhookTypeProw
-	}
 	if c.Cluster.Namespace == "" {
 		c.Cluster.Namespace = "jx"
 	}
-	if c.Cluster.GitKind == "" {
-		c.Cluster.GitKind = "github"
-	}
 	if c.Cluster.GitServer == "" {
 		c.Cluster.GitServer = "https://github.com"
 	}
+	if c.Cluster.GitKind == "" {
+		c.Cluster.GitKind = "github"
+	}
 	if c.Cluster.GitName == "" {
 		c.Cluster.GitName = c.Cluster.GitKind
 	}
 	if c.Ingress.NamespaceSubDomain == "" {
 		c.Ingress.NamespaceSubDomain = "-" + c.Cluster.Namespace + "."
 	}
+	if c.Webhook == WebhookTypeNone {
+		if c.Cluster.GitServer == "https://github.com" || c.Cluster.GitServer == "https://github.com/" {
+			c.Webhook = WebhookTypeProw
+		} else {
+			c.Webhook = WebhookTypeLighthouse
```

Comment:
I'm not sure this is a good idea without a strong warning to users that lighthouse is experimental. Can we add a confirmation dialog (override in batch mode) that says that we don't support their server type but that we do have an experimental feature called lighthouse that they could try out and make it really clear it's experimental!. Probably should be an error message to start with for the wrong type of git.

Reply:
this function has no way to know if its got batch mode enabled so gonna just omit defaulting for lighthouse for now until its GA